node/core/dispute-coordinator/src/real/initialized.rs

```diff
@@ -80,6 +85,11 @@ pub struct Initialized {
 	ordering_provider: OrderingProvider,
 	participation_receiver: WorkerMessageReceiver,
 	metrics: Metrics,
+	// This tracks only rolling session window failures.
+	// It can be a `Vec` if the need to track more arises.
+	error: Option<SessionsUnavailable>,
+	/// Latest relay blocks that have been succesfully scraped.
+	last_scraped_blocks: LruCache<Hash, ()>,
```

Comment:
Not entirely sure we want to model it this way, or rather track those `Hash`es that failed, and once their block number becomes too old, discard them. So a `BTreeMap<BlockNumber, Hash>` comes to mind.

Reply:
We could actually use a `bloom filter` here because of the space and time advantages over `a set` or any other data structure that could model a set.

Reply:
Hmm, not sure if a bloom filter is a good idea, there are no false negatives but we still could get false positives and not scrape a block which defeats the purpose of all this.

Reply:
This should work fine for transient errors, but would not work in practice for issues like the one we saw on Kusama. Also if we change the data structure here we can't reuse the `get_block_ancestors` fn so we'll duplicate more code. I think just increasing the cache size would be enough. The LRU semantics would help here as we always want to drop stuff that we no longer look because the rolling window advances.

Reply:
I'd like to make this deterministic, and an (expected) handfull of failed session info queries, should not be too compute intense on the node side.